lib/render/SliceRenderer.cpp

```diff
@@ -175,30 +176,58 @@ void SliceRenderer::_resetColormapCache()
     glTexImage1D(GL_TEXTURE_1D, 0, GL_RGBA8, _colorMapSize, 0, GL_RGBA, GL_FLOAT, &_cacheParams.tf_lut[0]);
 }
 
+// clang-format off
+
 int SliceRenderer::_regenerateSlice()
 {
     Grid *grid3d = nullptr;
     int   rc = _getGrid3D(grid3d);
     if (rc < 0) return -1;
 
     // Get data values from a slice
-    std::unique_ptr<float> dataValues(new float[_textureSideSize * _textureSideSize]);
+    std::shared_ptr<float> dataValues(new float[_textureSideSize * _textureSideSize]);
     planeDescription       pd;
     pd.sideSize = _textureSideSize;
     pd.origin = {_cacheParams.xOrigin, _cacheParams.yOrigin, _cacheParams.zOrigin};
     pd.rotation = {_cacheParams.xRotation, _cacheParams.yRotation, _cacheParams.zRotation};
     pd.boxMin = _cacheParams.boxMin;
     pd.boxMax = _cacheParams.boxMax;
-    pd.domainMin = _cacheParams.domainMin;
-    pd.domainMax = _cacheParams.domainMax;
-    RegularGrid *slice = SliceGridAlongPlane(grid3d, pd, dataValues, _windingOrder, _rectangle3D);
+
+    VAPoR::DimsType dims = { (size_t)_textureSideSize, (size_t)_textureSideSize, 1 };
+
+    ArbitrarilyOrientedRegularGrid* slice = new ArbitrarilyOrientedRegularGrid(
+        grid3d,
+        pd,
+        dims,
+        dataValues
+    );
+
+    CoordType corner1, corner2, corner3, corner4;
+    slice->GetUserCoordinates({0,0,0}, corner1);
+    slice->GetUserCoordinates({0,_textureSideSize,0}, corner2);
```

Comment:
Shouldn't all of these be indexed by `_textureSideSize-1`, not `_textureSizeSize`? I think the only reason that it is working as that this getter will clamp the indices to be in range.

Reply:
You're correct.  Fixed.